controllers/webhooks/component_webhook.go

```diff
@@ -86,13 +86,19 @@ func (r *ComponentWebhook) UpdateNudgedComponentStatus(ctx context.Context, obj
 
 		// Add the component to the status if it's not already present
 		if !util.StrInList(compName, nudgedComp.Status.BuildNudgedBy) {
-			nudgedComp.Status.BuildNudgedBy = append(nudgedComp.Status.BuildNudgedBy, compName)
 
 			// Update the Component's status - retry on conflict
-			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
-				err = r.client.Status().Update(ctx, nudgedComp)
+			retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Should `UpdateNudgedComponentStatus` return an error if `retry.RetryOnConflict(...) != nil`?

Reply:
Fixed. 

We don't want the webhook to error out here, so logging the error is sufficient.